include/Hdlcpp.hpp

```diff
@@ -185,7 +186,7 @@ class Hdlcpp {
         ControlTypeSelectiveReject,
     };
 
-    int encode(Frame &frame, uint8_t &sequenceNumber, const uint8_t *source, uint16_t sourceLength, std::vector<uint8_t> &destination)
+    size_t encode(Frame &frame, uint8_t &sequenceNumber, const std::span<const uint8_t> source, std::vector<uint8_t> &destination)
```

Comment:
size_t and -EINVAL does not match

Reply:
reverted to int to allow negative return codes